services/graph.js

```diff
@@ -1,136 +1,227 @@
+  if (tags.length > 0) {
+    return WRIAPI.put(`dataset/${datasetId}/vocabulary`,
+      {
+        knowledge_graph: {
+          tags,
+          application: process.env.APPLICATIONS,
+          env: process.env.API_ENV
+        }
+      },
+      { headers: { Authorization: token } })
+      .then(response => WRISerializer(response.data))
+      .catch((response) => {
+        const { status, statusText } = response;
+        logger.error(`Error updating dataset tags ${datasetId}: ${status}: ${statusText}`);
+        throw new Error(`Error updating dataset tags ${datasetId}: ${status}: ${statusText}`);
+      });
   }
-
-  /**
-  * Update dataset tags
-  */
-  updateDatasetTags(datasetId, tags, token, usePatch = false) {
-    let bodyObj = {
-      knowledge_graph: {
-        tags,
+  return WRIAPI.delete(`dataset/${datasetId}/vocabulary/knowledge_graph`,
+    {
+      headers: { Authorization: token },
+      params: {
         application: process.env.APPLICATIONS,
         env: process.env.API_ENV
       }
-    };
-    let method = tags.length > 0 ? 'PUT' : 'DELETE';
-    let url = `${process.env.WRI_API_URL}/dataset/${datasetId}/vocabulary`;
-
-    if (usePatch) {
-      method = 'PATCH';
-      bodyObj = { tags, application: process.env.APPLICATIONS };
-      url = `${url}/knowledge_graph`;
-    }
-
-    if (method === 'DELETE') {
-      url = `${url}/knowledge_graph?application=${process.env.APPLICATIONS}`;
-      bodyObj = {};
-    }
-
-    return fetch(url, {
-      method,
-      body: JSON.stringify(bodyObj),
-      headers: {
-        'Content-Type': 'application/json',
-        Authorization: token
-      }
     })
-      .then(response => response.json())
-      .then(jsonData => jsonData.data);
-  }
-
-  /**
-   * Send a request to count a view to the dataset
-   * @param {string} datasetId Dataset ID
-   * @param {string} [token] User token
-   * @returns {Promise<void>}
-   */
-  countDatasetView(datasetId, token) {
-    const headers = {};
-
-    if (token) {
-      headers.Authorization = token;
-    }
-
-    return fetch(`${process.env.WRI_API_URL}/graph/dataset/${datasetId}/visited?application=${process.env.APPLICATIONS}&env=${process.env.API_ENV}`, {
-      method: 'POST',
-      headers
+    .then(response => WRISerializer(response.data))
+    .catch((response) => {
+      const { status, statusText } = response;
+      logger.error(`Error updating dataset tags ${datasetId}: ${status}: ${statusText}`);
+      throw new Error(`Error updating dataset tags ${datasetId}: ${status}: ${statusText}`);
+    });
+};
+
+/**
+ * Send a request to count a view to the dataset
+ * @param {string} datasetId Dataset ID
+ * @param {string} [token] User token
+ * @returns {Promise<void>}
+ */
+export const countDatasetView = (datasetId, token, params = {
+  env: process.env.API_ENV,
+  application: process.env.APPLICATIONS
+}) => {
+  logger.info('Count dataset view');
+  return WRIAPI.post(`graph/dataset/${datasetId}/visited`,
+    {},
```

Comment:
Shouldn't  `params` object go here?

Reply:
Not really, it also caught my attention when I was refactoring it. However it seems that it's a `POST` endpoint that doesn't expect a body _(I think that the only information that it needs is gathered from the url itself and the user that is performing it)_